source/common/network/address_impl.cc

```diff
@@ -300,6 +358,34 @@ PipeInstance::PipeInstance(const std::string& pipe_path, mode_t mode,
   pipe_.mode_ = mode;
 }
 
+PipeInstance::PipeInstance(absl::Status& error, const sockaddr_un* address, socklen_t ss_len,
+                           mode_t mode, const SocketInterface* sock_interface)
+    : InstanceBase(Type::Pipe, sockInterfaceOrDefault(sock_interface)) {
+  if (address->sun_path[0] == '\0') {
+#if !defined(__linux__)
```

Comment:
factor out common code from two PipeInstance ctors using the same pattern as your ipv4/ipv6 ones. These are identical except on throws and the other returns the error, right?